html/includes/forms/save-host-dependency.inc.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2017 Aldemir Akpinar <https://github.com/aldemira/>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+if (is_admin() === false) {
+    $status = array('status' => 1, 'message' => 'You need to be admin');
+} else {
+    foreach ($_POST['parent_ids'] as $parent) {
+        if (!is_numeric($parent)) {
+            $status = array('status' => 1, 'message' => 'Parent ID must be an integer!');
+            break;
+        }
+    }
+
+    if (count($_POST['parent_ids']) > 1 && in_array('0', $_POST['parent_ids'])) {
+        $status = array('status' => 1, 'message' => 'Multiple parents cannot contain None-Parent!');
+    }
+
+    // A bit of an effort to reuse this code with dependency editing and the dependency wizard (editing multiple hosts at the same time)
+    $device_arr = array();
+    foreach ($_POST['device_ids'] as $dev) {
+        if (!is_numeric($dev)) {
+            $status = array('status' => 1, 'message' => 'Device ID must be an integer!');
+            break;
+        } elseif (in_array($dev, $_POST['parent_ids'])) {
+            $status = array('status' => 1, 'message' => 'A device cannot depend itself');
+            break;
+        }
+        foreach ($_POST['parent_ids'] as $parent) {
+            if (is_numeric($parent) && $parent != 0) {
+                dbInsert(array('parent_device_id' => $parent, 'child_device_id' => $dev), 'device_relationships');
```

Comment:
This doesn't take into account the removed parents. You might need to delete all parents associated first then re-populate them. You can also use dbBulkInsert to save on the loop.

Reply:
@laf OK, but we'll still need to validate the input before deleting anything, don't we? and that requires a for loop :(

Reply:
If you build up for the bulk insert you can then delete once you know you have valid data